### PR TITLE
Add /address/:address/cardinals API

### PR DESCRIPTION
**Pull request by lifofifoX**
This would make it easier for apps to rely on the address index for retrieving cardinals UTXOs.

### CODE REVIEW

File: src/subcommand/server.rs [hunk 1]
```diff
@@ -887,6 +888,56 @@ impl Server {
...
+  ) -> ServerResult {
+    task::block_in_place(|| {
+      Ok(if accept_json {
+        let address = address
+          .require_network(server_config.chain.network())
+          .map_err(|err| ServerError::BadRequest(err.to_string()))?;
+
+        let mut outputs = index.get_address_info(&address)?;
+
+        outputs.sort();
```
Comment (raphjaph) [inline src/subcommand/server.rs:905]: Do people rely on lexicographical ordering? Or should it be ordered by size?

File: tests/json_api.rs [hunk 2]
```diff
@@ -362,6 +362,54 @@ fn get_output() {
...
+          .unwrap()
+          .assume_checked()
+      ),
+      spent: false,
+      transaction: send.txid,
+      value: 2 * COIN_VALUE,
+    }]
+  );
+}
+
```
Comment (raphjaph) [inline tests/json_api.rs:412]: Would be good to add a test that asserts that runic and inscription utxos are not returned.

**Review by raphjaph — CHANGES_REQUESTED** [review]
Looks good so far! Just missing a test

File: src/subcommand/server.rs (hunk 1 above)
Comment (lifofifoX) [reply src/subcommand/server.rs:905]: I wasn't sure either! But followed the precedence of `address`

https://github.com/ordinals/ord/blob/f1c1ee5adb39c54e99c79e65791b6339186c35be/src/subcommand/server.rs#L860

File: tests/json_api.rs (hunk 2 above)
Comment (lifofifoX) [reply tests/json_api.rs:412]: Makes sense, will add them!

Do you think we should have `/address/:address/ordinals` endpoint for fetching just inscriptions and runes? I'm not sure about naming it `/ordinals` if it includes Runes, but it goes well with `/cardinals`.

**Comment by raphjaph** [comment]
Let me know when this PR is ready for review

File: tests/json_api.rs (hunk 2 above)
Comment (lifofifoX) [reply tests/json_api.rs:412]: Alright, I'll work on separate PRs.

**Comment by lifofifoX** [comment]
@raphjaph Added an additional test to check inscriptions/runes are excluded in cardinals response. Should be ready now!

**Comment by lifofifoX** [comment]
@raphjaph Added other two endpoints in #4008.

**Comment by raphjaph** [comment]
Let me have a look at this

**Comment by raphjaph** [comment]
I think we should go with something like this: https://github.com/ordinals/ord/pull/4056

What do you think?

**Comment by raphjaph** [comment]
Close by #4056

**Pull request closed without merging**